larsim/LegacyLArG4/OpFastScintillation.cxx

```diff
@@ -1059,36 +1052,36 @@ namespace larg4 {
   void OpFastScintillation::propagationTime(std::vector<double>& arrival_time_dist, G4ThreeVector x0,
                                              const size_t OpChannel, bool Reflected) //const
   {
-    static art::ServiceHandle<phot::PhotonVisibilityService const> pvs;
-    if (pvs->IncludeParPropTime() && pvs->IncludePropTime()) {
+    assert(fPVS);
+    if (fPVS->IncludeParPropTime() && fPVS->IncludePropTime()) {
       throw cet::exception("OpFastScintillation") << "Cannot have both propagation time models simultaneously.";
     }
-    else if (pvs->IncludeParPropTime() && !(ParPropTimeTF1  && (ParPropTimeTF1[OpChannel].GetNdim() == 1)) ) {
+    else if (fPVS->IncludeParPropTime() && !(ParPropTimeTF1  && (ParPropTimeTF1[OpChannel].GetNdim() == 1)) ) {
       //Warning: TF1::GetNdim()==1 will tell us if the TF1 is really defined or it is the default one.
       //This will fix a segfault when using timing and interpolation.
       G4cout << "WARNING: Requested parameterized timing, but no function found. Not applying propagation time." << G4endl;
     }
-    else if (pvs->IncludeParPropTime()) {
+    else if (fPVS->IncludeParPropTime()) {
       if (Reflected)
         throw cet::exception("OpFastScintillation") << "No parameterized propagation time for reflected light";
       for(size_t i = 0; i < arrival_time_dist.size(); ++i) {
         arrival_time_dist[i] = ParPropTimeTF1[OpChannel].GetRandom();
       }
     }
-    else if (pvs->IncludePropTime()) {
+    else if (fPVS->IncludePropTime()) {
       // Get VUV photons arrival time distribution from the parametrization
-      const G4ThreeVector OpDetPoint(fOpDetCenter.at(OpChannel)[0]*CLHEP::cm,
-                                     fOpDetCenter.at(OpChannel)[1]*CLHEP::cm,
-                                     fOpDetCenter.at(OpChannel)[2]*CLHEP::cm);
+      const G4ThreeVector OpDetPoint(fOpDetCenter.at(OpChannel).X()*CLHEP::cm,
+                                     fOpDetCenter.at(OpChannel).Y()*CLHEP::cm,
+                                     fOpDetCenter.at(OpChannel).Z()*CLHEP::cm);
```

Comment:
Clearer, yes.  For efficiency reasons, I'd prefer something like:
```.cpp
auto const& point = fOpDetCenter.at(OpChannel);
const G4ThreeVector OpDetPoint(point.X()*CLHEP::cm,
                               point.Y()*CLHEP::cm,
                               point.Z()*CLHEP::cm);
```
No reason to check the range more than once--and I know you didn't introduce the multiple  `at(...)` calls.

Reply:
I overlooked a lot of good practice issues... including this one.
Since I have to merge it again, I'll fix this as well.